channeldb/migrations.go

```diff
@@ -746,9 +746,14 @@ func migrateGossipMessageStoreKeys(tx *bbolt.Tx) error {
 // payments) we delete those statuses, so only Completed payments remain in the
 // new bucket structure.
 func migrateOutgoingPayments(tx *bbolt.Tx) error {
-	oldPayments, err := tx.CreateBucketIfNotExists(paymentBucket)
-	if err != nil {
-		return err
+	log.Infof("Migrating outgoing payments to new bucket structure")
+
+	oldPayments := tx.Bucket(paymentBucket)
+
+	// Return early if there are no payments to migrate.
+	if oldPayments == nil || oldPayments.Stats().KeyN == 0 {
```

Comment:
Historically we've avoided using the `Stats` since it can have undefined behavior in some cases

Reply:
Probably best to avoid then. Pushed fixup with a diff approach.